tests/test_x509_ext.py

```diff
@@ -1578,6 +1615,30 @@ def test_invalid_rfc822name(self, backend):
 
         assert 'Invalid rfc822name value' in str(exc.value)
 
+    def test_other_name(self, backend):
+        cert = _load_cert(
+            os.path.join(
+                "x509", "custom", "san_other_name.pem"
+            ),
+            x509.load_pem_x509_certificate,
+            backend
+        )
+
+        ext = cert.extensions.get_extension_for_oid(
+            x509.OID_SUBJECT_ALTERNATIVE_NAME
+        )
+        assert ext is not None
+        assert ext.critical is False
+
+        assert len(ext.value) == 1
+        assert list(ext.value)[0] == \
+            x509.OtherName(
+                x509.ObjectIdentifier("1.2.3.4"),
+                b'\x16\x0bHello World')
+
+        othernames = ext.value.get_values_for_type(x509.OtherName)
+        assert othernames == [b'\x16\x0bHello World']
```

Comment:
Meaning `.type_id` an OID, `.<something creative>` be the der data, and
`.value` be `self`? I'm ok with that.

On Sun, Jul 5, 2015 at 5:56 PM, Joshua Tauberer <notifications@github.com>
wrote:

> In tests/test_x509_ext.py
> <https://github.com/pyca/cryptography/pull/2108#discussion_r33899392>:
>
> > +        )
> > +
> > +        ext = cert.extensions.get_extension_for_oid(
> > +            x509.OID_SUBJECT_ALTERNATIVE_NAME
> > +        )
> > +        assert ext is not None
> > +        assert ext.critical is False
> > +
> > +        assert len(ext.value) == 1
> > +        assert list(ext.value)[0] == \
> > +            x509.OtherName(
> > +                x509.ObjectIdentifier("1.2.3.4"),
> > +                b'\x16\x0bHello World')
> > +
> > +        othernames = ext.value.get_values_for_type(x509.OtherName)
> > +        assert othernames == [b'\x16\x0bHello World']
>
> Could also be done by having value return self in OtherName?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/2108/files#r33899392>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084